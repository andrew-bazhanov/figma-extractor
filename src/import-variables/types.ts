import type { FileNodesResponse, FullStyleMetadata } from 'figma-js';

export interface ImportVariablesAdapter {
  readonly data: unknown;
  get styleMetadata(): FullStyleMetadata[];
  get fileNodes(): FileNodesResponse;
}

export interface ImportVariablesValidatorData {
  validateData: () => Error | undefined;
}

export type DeepWriteable<T> = { -readonly [P in keyof T]: DeepWriteable<T[P]> };
