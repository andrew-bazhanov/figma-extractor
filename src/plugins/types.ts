import { Core, FigmaData } from '../core';

export interface Plugin {
  (core: Core, data: FigmaData): Promise<void> | void;
  pluginName: string;
}
