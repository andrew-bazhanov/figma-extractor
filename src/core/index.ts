import { ClientInterface, FileNodesResponse, FullStyleMetadata } from 'figma-js';

import { Plugin } from '../plugins';
import { runFormattingFile } from './run-formatting-file';
import { styleTypeUtils } from './style-type-utils';
import { writeJsFile } from './write-js-file';

export interface FigmaData {
  figmaClient: ClientInterface;
  styleMetadata: readonly FullStyleMetadata[];
  fileNodes: FileNodesResponse;
}

export interface Core {
  config: Config;
  rootPath: string;
  plugins: Plugin[];
  styleTypeUtils: typeof styleTypeUtils;
  writeJsFile: typeof writeJsFile;
  runFormattingFile: typeof runFormattingFile;
}

export const createCore = (args: { config: Config; rootPath: string; plugins: Plugin[] }): Core => {
  return {
    ...args,
    styleTypeUtils,
    writeJsFile,
    runFormattingFile: runFormattingFile,
  };
};
