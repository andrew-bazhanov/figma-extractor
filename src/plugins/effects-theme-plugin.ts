import path from 'path';

import { Mode } from '@/types';

import { getEffectStyles } from '../lib/get-effect-styles';
import { stringifyRecordsWithSort } from '../lib/stringify';
import {
  addNewNameToEachTheme,
  checkConfigAndThrowCommonError,
  createThemeCollection,
  generateCSSVariables,
  generateJsColors,
  generateJsVariables,
  generateThemeListTS,
  getRealThemesFromCollection,
  NOT_FOUND_THEME_NAME,
  separateThemes,
  variableNameIsValid,
} from '../lib/themes';
import { Plugin } from './types';

type EffectName = string;

interface FormattedEffects {
  backdropBlur?: {
    [key: string]: string;
  };
  [key: string]: any; // This allows for any other key-value pairs
}

const formattedEffects = (effects: Record<string, string>): FormattedEffects => {
  return Object.entries(effects).reduce((acc, [name, value]) => {
    if (name.includes('bg-blur')) {
      const backdropBlur = acc.backdropBlur || {};
      // format name from bg-blur-100 to 100
      // because className in tailwind backdrop-blur-100 better then backdrop-blur-bg-blur-100
      const backdropBlurName = name.split('-')[2];

      return {
        ...acc,
        backdropBlur: {
          ...backdropBlur,
          [backdropBlurName]: value,
        },
      };
    }

    return { ...acc, [name]: value };
  }, {} as FormattedEffects);
};

export const effectsThemePlugin: Plugin = (
  { config, writeFile, addEslintDisableRules, log },
  { variables },
) => {
  if (config?.styles?.responsive?.disabled) {
    log(
      '[info:effects-theme] >>> ',
      'Effects-theme plugin has been disabled so the plugin had not been launched',
    );

    return;
  }
  log('[info:effects-theme] >>> ', 'Effects-theme plugin starts working...');
  const filteredCollections = variables.filter(({ name }) =>
    config?.styles?.effects?.collectionNames?.includes(name),
  );

  const modes = filteredCollections.reduce<Mode[]>((acc, item) => [...acc, ...item.modes], []);

  const { allowedThemes, defaultTheme } = checkConfigAndThrowCommonError({
    allowedThemes: config?.styles?.allowedThemes,
    defaultTheme: config?.styles?.defaultTheme,
  });

  log('[info:effects-theme] >>> ', 'Allowed themes: ', JSON.stringify(allowedThemes));
  log('[info:effects-theme] >>> ', 'Default themes: ', defaultTheme);

  const themesCollection = createThemeCollection({ allowedThemes });
  const effects = getEffectStyles(modes, config);

  let anyThemeIsUsed = false;
  for (const { newName, value, theme, originalName } of separateThemes({
    allowedThemes,
    data: effects,
  })) {
    log(
      '[info:effects-theme] >>> ',
      `'${newName}' ${
        theme === NOT_FOUND_THEME_NAME ? 'without theme' : `on '${theme}' theme`
      } => '${value}'  (original: '${originalName}')`,
    );
    if (theme === NOT_FOUND_THEME_NAME) {
      if (!variableNameIsValid(newName)) {
        throw new Error(`Effect name: "${newName}" without theme contains not-valid chars.`);
      }
      themesCollection[NOT_FOUND_THEME_NAME][newName] = value;
    } else {
      if (!variableNameIsValid(newName)) {
        throw new Error(
          `Effect name: "${newName}" from "${theme}" theme contains not-valid chars.`,
        );
      }

      anyThemeIsUsed = true;
      addNewNameToEachTheme(themesCollection, newName);
      themesCollection[theme][newName] = value;
    }
  }

  if (!anyThemeIsUsed) {
    throw new Error(
      "None of themes was found inside figma data. Check your themes and figma's themes",
    );
  }

  for (const { themeName, variables, currentThemeIsDefault } of getRealThemesFromCollection({
    themesCollection,
    defaultTheme,
  })) {
    const jsData: Record<EffectName, string> = generateJsColors(variables);
    const { backdropBlur, ...formattedData } = formattedEffects(jsData);
    for (const [name, value] of Object.entries(jsData)) {
      log(`[info:effects-theme/${themeName}] >>> `, `'${name}' => '${value}'`);
    }

    const cssData = generateCSSVariables(variables, themeName);

    const fullPath = path.join(config?.styles?.exportPath || '', `effects/${themeName}`);

    const jsTemplate = `module.exports = {boxShadow: ${stringifyRecordsWithSort(formattedData)},
    ${backdropBlur ? `backdropBlur: ${stringifyRecordsWithSort(backdropBlur)}` : ''}}`;

    writeFile(
      addEslintDisableRules(jsTemplate, [
        'disable-max-lines',
        'disable-typescript-naming-convention',
      ]),
      path.join(fullPath, 'index.js'),
    );

    writeFile(cssData, path.join(fullPath, 'vars.css'));

    if (currentThemeIsDefault) {
      const fullPath = path.join(config?.styles?.exportPath || '', 'effects');

      const jsData = generateJsVariables(
        variables,
        defaultTheme && themesCollection[defaultTheme] ? themesCollection[defaultTheme] : {},
      );

      const { backdropBlur, ...formattedData } = formattedEffects(jsData);
      const jsTemplate = `module.exports = {boxShadow: ${stringifyRecordsWithSort(formattedData)},
      ${backdropBlur ? `backdropBlur: ${stringifyRecordsWithSort(backdropBlur)}` : ''}}`;

      writeFile(
        addEslintDisableRules(jsTemplate, [
          'disable-max-lines',
          'disable-typescript-naming-convention',
        ]),
        path.join(fullPath, 'with-vars.js'),
      );

      const jsLegacyData = generateJsColors(variables);
      const { backdropBlur: backdropBlurLegacy, ...formattedDataLegacy } =
        formattedEffects(jsLegacyData);

      const jsLegacyTemplate = `module.exports = {boxShadow: ${stringifyRecordsWithSort(formattedDataLegacy)},
      ${backdropBlurLegacy ? `backdropBlur: ${stringifyRecordsWithSort(backdropBlurLegacy)}` : ''}}`;

      writeFile(
        addEslintDisableRules(jsLegacyTemplate, [
          'disable-max-lines',
          'disable-typescript-naming-convention',
        ]),
        path.join(fullPath, 'index.js'),
      );

      const cssData = generateCSSVariables(variables);
      writeFile(cssData, path.join(fullPath, 'vars.css'));
    }
  }

  const themeListTSData = generateThemeListTS(themesCollection, defaultTheme);

  writeFile(
    addEslintDisableRules(themeListTSData, [
      'disable-max-lines',
      'disable-typescript-naming-convention',
    ]),
    path.join(config?.styles?.exportPath || '', 'themes-list.ts'),
  );
};

effectsThemePlugin.pluginName = 'effects-theme';
