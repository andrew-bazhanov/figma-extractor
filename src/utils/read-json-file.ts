import fs from 'fs';

export function readJsonFile<T = any>(filePath: string): T {
  try {
    const jsonData = fs.readFileSync(filePath, 'utf-8');

    return JSON.parse(jsonData);
  } catch (error) {
    console.error('Error reading the JSON file:', error);
    throw error; // rethrow the error for further handling if needed
  }
}
