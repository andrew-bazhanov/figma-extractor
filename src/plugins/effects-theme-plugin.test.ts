/* eslint-disable max-lines */
import path from 'path';

import { createCore } from '../core';
import { ThemeVariablesConfig } from '../types';
import { fileNodes } from '../utils/generate-styles/fixtures/file-nodes';
import { styleMetadata } from '../utils/generate-styles/fixtures/style-metadata';
import { readJsonFile } from '../utils/read-json-file';
import { effectsThemePlugin } from './effects-theme-plugin';

describe('effectsThemePlugin', () => {
  const rootPath = process.cwd();
  const jsonVariablesPath = './variables.json';
  const variables = readJsonFile<ThemeVariablesConfig[]>(path.join(rootPath, jsonVariablesPath));

  it('should throw the error because the allowedThemes is not defined', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow(
      '`config -> styles -> allowedThemes` field is required when the useTheme is equal true',
    );
  });

  it('should throw the error because the defaultTheme is not defined', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: [],
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow(
      '`config -> styles -> defaultTheme` field is required when the useTheme is equal true',
    );
  });

  it('should throw the error because the allowedThemes is empty', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: [],
          defaultTheme: 'any',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow('`config -> styles -> allowedThemes` field must have one or more theme name');
  });

  it('should throw the error because the defined defaultTheme is not included in allowedThemes list', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light', 'dark', 'monochrome'],
          defaultTheme: 'another',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow("`config -> styles -> defaultTheme` field must be one of allowedThemes' values");
  });

  it("should throw the error because any theme wasn't found inside figma data", () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['theme1', 'theme2'],
          defaultTheme: 'theme1',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow("None of themes was found inside figma data. Check your themes and figma's themes");
  });

  it('should throw the error because some colors from themes contain not-valid chars', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light', 'dark', 'monochrome'],
          defaultTheme: 'monochrome',
          effects: {
            collectionNames: ['effects'],
            keyName: (name?: string) => {
              return name === 'light/shadow-600' ? name + '+prefix' : name;
            },
          },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow('Effect name: "shadow-600+prefix" from "light" theme contains not-valid chars.');
  });

  it('should throw the error because some colors from no themes contain not-valid chars', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light'],
          defaultTheme: 'light',
          effects: {
            collectionNames: ['effects'],

            keyName: (name?: string) => {
              return name === 'dark/shadow-600' ? name + '+prefix' : name;
            },
          },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    expect(() =>
      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any),
    ).toThrow('Effect name: "dark-shadow-600+prefix" without theme contains not-valid chars.');
  });

  it('should create js files with css variables', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light', 'dark', 'monochrome'],
          defaultTheme: 'monochrome',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    effectsThemePlugin(core, { styleMetadata: styleMetadata.styles, fileNodes, variables } as any);

    expect(core.writeFile).toHaveBeenCalled();
    expect((core.writeFile as jest.Mock).mock.calls[0][1]).toBe(
      '/export-path/effects/light/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[0][0]).toMatchSnapshot(
      '/export-path/effects/light/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[2][1]).toBe(
      '/export-path/effects/dark/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[2][0]).toMatchSnapshot(
      '/export-path/effects/dark/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[4][1]).toBe(
      '/export-path/effects/monochrome/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[4][0]).toMatchSnapshot(
      '/export-path/effects/monochrome/index.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[6][1]).toBe(
      '/export-path/effects/with-vars.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[6][0]).toMatchSnapshot(
      '/export-path/effects/with-vars.ts',
    );
    expect((core.writeFile as jest.Mock).mock.calls[7][1]).toBe('/export-path/effects/index.ts');
    expect((core.writeFile as jest.Mock).mock.calls[7][0]).toMatchSnapshot(
      '/export-path/effects/index.ts',
    );
  });

  it('should create css files with css variables', () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light', 'dark', 'monochrome'],
          defaultTheme: 'monochrome',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    effectsThemePlugin(core, { styleMetadata: styleMetadata.styles, fileNodes, variables } as any);

    expect(core.writeFile).toHaveBeenCalled();
    expect((core.writeFile as jest.Mock).mock.calls[1][1]).toBe(
      '/export-path/effects/light/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[1][0]).toMatchSnapshot(
      '/export-path/effects/light/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[3][1]).toBe(
      '/export-path/effects/dark/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[3][0]).toMatchSnapshot(
      '/export-path/effects/dark/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[5][1]).toBe(
      '/export-path/effects/monochrome/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[5][0]).toMatchSnapshot(
      '/export-path/effects/monochrome/vars.css',
    );
    expect((core.writeFile as jest.Mock).mock.calls[8][1]).toBe('/export-path/effects/vars.css');
    expect((core.writeFile as jest.Mock).mock.calls[8][0]).toMatchSnapshot(
      '/export-path/effects/vars.css',
    );
  });

  it("should create theme-list.ts files with themes' types", () => {
    const core = createCore({
      config: {
        styles: {
          exportPath: '/export-path/',
          allowedThemes: ['light', 'dark', 'monochrome'],
          defaultTheme: 'monochrome',
          effects: { collectionNames: ['effects'] },
        },
      },
      plugins: [],
      rootPath: '/root-path',
      log: jest.fn(),
    });

    core.writeFile = jest.fn();

    effectsThemePlugin(core, { styleMetadata: styleMetadata.styles, fileNodes, variables } as any);

    expect(core.writeFile).toHaveBeenCalled();
    expect((core.writeFile as jest.Mock).mock.calls[9][1]).toBe('/export-path/themes-list.ts');
    expect((core.writeFile as jest.Mock).mock.calls[9][0]).toMatchSnapshot(
      '/export-path/themes-list.ts',
    );
  });

  describe("create theme data with one theme which doesn't exist in figma", () => {
    it('should create js files with css variables', () => {
      const core = createCore({
        config: {
          styles: {
            exportPath: '/export-path/',
            allowedThemes: ['light', 'dark-blue', 'monochrome'],
            defaultTheme: 'monochrome',
            effects: { collectionNames: ['effects'] },
          },
        },
        plugins: [],
        rootPath: '/root-path',
        log: jest.fn(),
      });

      core.writeFile = jest.fn();

      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any);

      expect(core.writeFile).toHaveBeenCalled();
      expect((core.writeFile as jest.Mock).mock.calls[0][1]).toBe(
        '/export-path/effects/light/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[0][0]).toMatchSnapshot(
        '/export-path/effects/light/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[2][1]).toBe(
        '/export-path/effects/dark-blue/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[2][0]).toMatchSnapshot(
        '/export-path/effects/dark-blue/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[4][1]).toBe(
        '/export-path/effects/monochrome/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[4][0]).toMatchSnapshot(
        '/export-path/effects/monochrome/index.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[6][1]).toBe(
        '/export-path/effects/with-vars.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[6][0]).toMatchSnapshot(
        '/export-path/effects/with-vars.ts',
      );
      expect((core.writeFile as jest.Mock).mock.calls[7][1]).toBe('/export-path/effects/index.ts');
      expect((core.writeFile as jest.Mock).mock.calls[7][0]).toMatchSnapshot(
        '/export-path/effects/index.ts',
      );
    });

    it('should create css files with css variables', () => {
      const core = createCore({
        config: {
          styles: {
            exportPath: '/export-path/',
            allowedThemes: ['light', 'dark-blue', 'monochrome'],
            defaultTheme: 'monochrome',
            effects: { collectionNames: ['effects'] },
          },
        },
        plugins: [],
        rootPath: '/root-path',
        log: jest.fn(),
      });

      core.writeFile = jest.fn();

      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any);

      expect(core.writeFile).toHaveBeenCalled();
      expect((core.writeFile as jest.Mock).mock.calls[1][1]).toBe(
        '/export-path/effects/light/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[1][0]).toMatchSnapshot(
        '/export-path/effects/light/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[3][1]).toBe(
        '/export-path/effects/dark-blue/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[3][0]).toMatchSnapshot(
        '/export-path/effects/dark-blue/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[5][1]).toBe(
        '/export-path/effects/monochrome/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[5][0]).toMatchSnapshot(
        '/export-path/effects/monochrome/vars.css',
      );
      expect((core.writeFile as jest.Mock).mock.calls[8][1]).toBe('/export-path/effects/vars.css');
      expect((core.writeFile as jest.Mock).mock.calls[8][0]).toMatchSnapshot(
        '/export-path/effects/vars.css',
      );
    });

    it("should create theme-list.ts files with themes' types", () => {
      const core = createCore({
        config: {
          styles: {
            exportPath: '/export-path/',
            allowedThemes: ['light', 'dark-blue', 'monochrome'],
            defaultTheme: 'monochrome',
            effects: { collectionNames: ['effects'] },
          },
        },
        plugins: [],
        rootPath: '/root-path',
        log: jest.fn(),
      });

      core.writeFile = jest.fn();

      effectsThemePlugin(core, {
        styleMetadata: styleMetadata.styles,
        fileNodes,
        variables,
      } as any);

      expect(core.writeFile).toHaveBeenCalled();
      expect((core.writeFile as jest.Mock).mock.calls[9][1]).toBe('/export-path/themes-list.ts');
      expect((core.writeFile as jest.Mock).mock.calls[9][0]).toMatchSnapshot(
        '/export-path/themes-list.ts',
      );
    });
  });
});
