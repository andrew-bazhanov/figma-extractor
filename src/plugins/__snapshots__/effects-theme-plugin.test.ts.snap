// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create css files with css variables: /export-path/effects/dark-blue/vars.css 1`] = `
"
[data-theme='dark-blue'] {
  --sh-bg-blur-100: '';
--sh-bg-blur-200: '';
--sh-shadow-100: '';
--sh-shadow-200: '';
--sh-shadow-300: '';
--sh-shadow-400: '';
--sh-shadow-500: '';
--sh-shadow-600: '';
}"
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create css files with css variables: /export-path/effects/light/vars.css 1`] = `
"
[data-theme='light'] {
  --sh-bg-blur-100: blur(8px);
--sh-bg-blur-200: blur(20px);
--sh-shadow-100: 0px 1px 0px rgba(0, 0, 0, 0.05);
--sh-shadow-200: 0px 1px 0px rgba(0, 0, 0, 0.15);
--sh-shadow-300: 0px 4px -1px rgba(0, 0, 0, 0.15);
--sh-shadow-400: 0px 16px -4px rgba(0, 0, 0, 0.20);
--sh-shadow-500: 0px 16px -8px rgba(0, 0, 0, 0.10);
--sh-shadow-600: 0px 16px -8px rgba(0, 0, 0, 0.40);
}"
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create css files with css variables: /export-path/effects/monochrome/vars.css 1`] = `
"
[data-theme='monochrome'] {
  --sh-bg-blur-100: '';
--sh-bg-blur-200: '';
--sh-shadow-100: '';
--sh-shadow-200: '';
--sh-shadow-300: '';
--sh-shadow-400: '';
--sh-shadow-500: '';
--sh-shadow-600: '';
}"
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create css files with css variables: /export-path/effects/vars.css 1`] = `
"
:root {
  --sh-bg-blur-100: '';
--sh-bg-blur-200: '';
--sh-shadow-100: '';
--sh-shadow-200: '';
--sh-shadow-300: '';
--sh-shadow-400: '';
--sh-shadow-500: '';
--sh-shadow-600: '';
}"
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create js files with css variables: /export-path/effects/dark-blue/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {},
    }
    "
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create js files with css variables: /export-path/effects/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {},
      }
    "
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create js files with css variables: /export-path/effects/light/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {"shadow-100":"0px 1px 0px rgba(0, 0, 0, 0.05)","shadow-200":"0px 1px 0px rgba(0, 0, 0, 0.15)","shadow-300":"0px 4px -1px rgba(0, 0, 0, 0.15)","shadow-400":"0px 16px -4px rgba(0, 0, 0, 0.20)","shadow-500":"0px 16px -8px rgba(0, 0, 0, 0.10)","shadow-600":"0px 16px -8px rgba(0, 0, 0, 0.40)"},
    backdropBlur: {"100":"blur(8px)","200":"blur(20px)"}}
    "
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create js files with css variables: /export-path/effects/monochrome/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {},
    }
    "
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create js files with css variables: /export-path/effects/with-vars.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {"shadow-100":"var(--sh-shadow-100,'')","shadow-200":"var(--sh-shadow-200,'')","shadow-300":"var(--sh-shadow-300,'')","shadow-400":"var(--sh-shadow-400,'')","shadow-500":"var(--sh-shadow-500,'')","shadow-600":"var(--sh-shadow-600,'')"},
      backdropBlur: {"100":"var(--sh-bg-blur-100,'')","200":"var(--sh-bg-blur-200,'')"}}
    "
`;

exports[`effectsThemePlugin create theme data with one theme which doesn't exist in figma should create theme-list.ts files with themes' types: /export-path/themes-list.ts 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    
// THIS FILE IS GENERATED AUTOMATICALLY. DON'T CHANGE IT.

export const DEFAULT_THEME = 'monochrome';

export const THEMES = ['light', 'dark-blue', 'monochrome'] as const;  

export type Theme = typeof THEMES[number];
  
    "
`;

exports[`effectsThemePlugin should create css files with css variables: /export-path/effects/dark/vars.css 1`] = `
"
[data-theme='dark'] {
  --sh-bg-blur-100: blur(8px);
--sh-bg-blur-200: blur(20px);
--sh-shadow-100: 0px 0px 1px rgba(255, 255, 255, 0.10);
--sh-shadow-200: 0px 0px 1px rgba(255, 255, 255, 0.10);
--sh-shadow-300: 0px 0px 1px rgba(255, 255, 255, 0.10);
--sh-shadow-400: 0px 0px 1px rgba(255, 255, 255, 0.20);
--sh-shadow-500: 0px 0px 1px rgba(255, 255, 255, 0.20);
--sh-shadow-600: 0px 0px 1px rgba(255, 255, 255, 0.20);
}"
`;

exports[`effectsThemePlugin should create css files with css variables: /export-path/effects/light/vars.css 1`] = `
"
[data-theme='light'] {
  --sh-bg-blur-100: blur(8px);
--sh-bg-blur-200: blur(20px);
--sh-shadow-100: 0px 1px 0px rgba(0, 0, 0, 0.05);
--sh-shadow-200: 0px 1px 0px rgba(0, 0, 0, 0.15);
--sh-shadow-300: 0px 4px -1px rgba(0, 0, 0, 0.15);
--sh-shadow-400: 0px 16px -4px rgba(0, 0, 0, 0.20);
--sh-shadow-500: 0px 16px -8px rgba(0, 0, 0, 0.10);
--sh-shadow-600: 0px 16px -8px rgba(0, 0, 0, 0.40);
}"
`;

exports[`effectsThemePlugin should create css files with css variables: /export-path/effects/monochrome/vars.css 1`] = `
"
[data-theme='monochrome'] {
  --sh-bg-blur-100: '';
--sh-bg-blur-200: '';
--sh-shadow-100: '';
--sh-shadow-200: '';
--sh-shadow-300: '';
--sh-shadow-400: '';
--sh-shadow-500: '';
--sh-shadow-600: '';
}"
`;

exports[`effectsThemePlugin should create css files with css variables: /export-path/effects/vars.css 1`] = `
"
:root {
  --sh-bg-blur-100: '';
--sh-bg-blur-200: '';
--sh-shadow-100: '';
--sh-shadow-200: '';
--sh-shadow-300: '';
--sh-shadow-400: '';
--sh-shadow-500: '';
--sh-shadow-600: '';
}"
`;

exports[`effectsThemePlugin should create js files with css variables: /export-path/effects/dark/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {"shadow-100":"0px 0px 1px rgba(255, 255, 255, 0.10)","shadow-200":"0px 0px 1px rgba(255, 255, 255, 0.10)","shadow-300":"0px 0px 1px rgba(255, 255, 255, 0.10)","shadow-400":"0px 0px 1px rgba(255, 255, 255, 0.20)","shadow-500":"0px 0px 1px rgba(255, 255, 255, 0.20)","shadow-600":"0px 0px 1px rgba(255, 255, 255, 0.20)"},
    backdropBlur: {"100":"blur(8px)","200":"blur(20px)"}}
    "
`;

exports[`effectsThemePlugin should create js files with css variables: /export-path/effects/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {},
      }
    "
`;

exports[`effectsThemePlugin should create js files with css variables: /export-path/effects/light/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {"shadow-100":"0px 1px 0px rgba(0, 0, 0, 0.05)","shadow-200":"0px 1px 0px rgba(0, 0, 0, 0.15)","shadow-300":"0px 4px -1px rgba(0, 0, 0, 0.15)","shadow-400":"0px 16px -4px rgba(0, 0, 0, 0.20)","shadow-500":"0px 16px -8px rgba(0, 0, 0, 0.10)","shadow-600":"0px 16px -8px rgba(0, 0, 0, 0.40)"},
    backdropBlur: {"100":"blur(8px)","200":"blur(20px)"}}
    "
`;

exports[`effectsThemePlugin should create js files with css variables: /export-path/effects/monochrome/index.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {},
    }
    "
`;

exports[`effectsThemePlugin should create js files with css variables: /export-path/effects/with-vars.js 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    module.exports = {boxShadow: {"shadow-100":"var(--sh-shadow-100,'')","shadow-200":"var(--sh-shadow-200,'')","shadow-300":"var(--sh-shadow-300,'')","shadow-400":"var(--sh-shadow-400,'')","shadow-500":"var(--sh-shadow-500,'')","shadow-600":"var(--sh-shadow-600,'')"},
      backdropBlur: {"100":"var(--sh-bg-blur-100,'')","200":"var(--sh-bg-blur-200,'')"}}
    "
`;

exports[`effectsThemePlugin should create theme-list.ts files with themes' types: /export-path/themes-list.ts 1`] = `
"
    /* eslint-disable max-lines */
    /* eslint-disable @typescript-eslint/naming-convention */
    
// THIS FILE IS GENERATED AUTOMATICALLY. DON'T CHANGE IT.

export const DEFAULT_THEME = 'monochrome';

export const THEMES = ['light', 'dark', 'monochrome'] as const;  

export type Theme = typeof THEMES[number];
  
    "
`;
